### DISCUSSION TITLE
Using different property instead of "id" for primary key

### DISCUSSION THREAD

**Discussion by bryan-gc** [mikro-orm/mikro-orm#3557] [Q&A] [answered]
Hi, how can I use a different property for my primary key besides "id", for example "areaId". I notice that when I use something different to "id" I need to specify it in OptionalProps `[OptionalProps]? : 'areaId' ` to allow me to use repository.create() without needing to define the autoincrementable id, and also when I want to get a reference with a repository typescript gives me this error

```
const areRef = this.areaRepository.getReference(1);
Argument of type 'number' is not assignable to parameter of type 'never[]'
```
![image](https://user-images.githubusercontent.com/73864592/193421279-305561f2-03d2-47f3-9aab-8b399352fb09.png)

but when I change the property to id no error is shown.

![image](https://user-images.githubusercontent.com/73864592/193421285-9eecb29f-af9d-4f24-a176-643a1b751fbb.png)

**Comment by B4nan** [comment, marked as answer]
You can use `PrimaryKeyType` and `PrimaryKeyProp` symbols, that way you pass the information about your PK to the TS level - as there is no way TS can tell the property type based on decorator.

```ts
class MyEntity {

  @PrimaryKey()
  foo: number;

  [PrimaryKeyProp]?: 'foo'; // this tells TS that foo is the primary key
  [PrimaryKeyType]?: number; // this tells TS that foo is of type number

}
```

The `PrimaryKeyProp` should be enough for you, if not, specify both. 

The ORM will automatically detect PKs under `id/_id/uuid` properties, so you only need this for composite PKs or in general when using property with different name.

**Reply by bryan-gc** [reply]
Thanks, that's what I needed.
